src/Tools/dotnet-monitor/Extensibility/FolderExtensionRepository.cs

```diff
@@ -28,12 +29,23 @@ public override bool TryFindExtension(string extensionName, out IExtension exten
 
             foreach (IFileInfo extensionDir in extensionDirs)
             {
-                if (_fileSystem.TryGetExtensionDefinitionPath(extensionDir.Name, out string definitionPath))
+                if (extensionDir.IsDirectory && !string.IsNullOrEmpty(extensionDir.PhysicalPath))
                 {
-                    var currExtension = new ProgramExtension(extensionName, _targetFolder, _fileSystem, definitionPath, _logger);
-                    if (extensionName == currExtension.Declaration.Name)
+                    string manifestPath = Path.Combine(extensionDir.PhysicalPath, ExtensionManifest.DefaultFileName);
+
+                    ExtensionManifest manifest;
+                    try
+                    {
+                        manifest = ExtensionManifest.FromPath(manifestPath);
+                    }
+                    catch (Exception ex) when (LogManifestParseError(ex, manifestPath))
                     {
-                        extension = currExtension;
+                        continue;
+                    }
+
+                    if (extensionName == manifest.Name)
```

Comment:
Do we have any case-sensitivity or restrictions on extension names that should be reflected here?

Reply:
I don't think that that was explicitly decided, but I think it's correct to enforce case sensitivity here. This name is not used in the file system anymore (e.g. you can name the extension's folder `xyz` but the manifest has `abc`; the `abc` is what will be expected in configuration and the `xyz` folder name has no bearing on the extension).